### PR TITLE
On Linux build with clang as well

### PR DESCRIPTION
**Pull request by pmatos**
On Push CI build with `clang` besides gcc.
This is one step towards replacing the current testing on travis - next up is the configure flags matrix.

### CODE REVIEW

File: .github/workflows/ci-push.yml [hunk 1]
```diff
@@ -121,9 +127,11 @@ jobs:
       working-directory: ./racket/src
       run: make -j  $((cpus+1)) install
     - name: Tarballing
+      if: matrix.cc == 'gcc' || matrix.os == 'macos-latest'
```
Comment (samth) [inline .github/workflows/ci-push.yml]: Why do we explicitly need to test this here?

Comment (pmatos) [reply .github/workflows/ci-push.yml]: We do not need to test racket build with `gcc` and `clang`. We build with `clang` but do not test the build. We test the `gcc` one. So we only tarball the `gcc` version. At the same time, `macos` only builds with `clang`, so we tarball in that case as well.

Happy to change this if you think it's worth it.

Comment (samth) [reply .github/workflows/ci-push.yml]: That seems fine (the odds of finding a miscompilation in clang that causes test failures is fine). Can you add a comment to that line about why?

**Comment by pmatos** [comment]
@samth requested comment clarification added at https://github.com/racket/racket/commit/b14b2400fd3ebf693fee53b144063f4fd6473b47

**Pull request merged**